### PR TITLE
add MITx Users model

### PR DESCRIPTION
**Pull request by rachellougee**
## Description

It adds int__mitx__users model, which combines MITx Online and edX.org users. 
It filters out those users from old MITProfessionalX and xPro courses

## Motivation and Context

https://github.com/mitodl/ol-data-platform/issues/558

## How Has This Been Tested?

ran dbt run on staging and intermediate models, and dbt test

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Enhancement (improves on existing behavior)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project. (Did you install and run the pre-commit hooks?)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+, mitx_users as (
+    select
+        '{{ var("mitxonline") }}' as platform
+        , user_id
+        , user_username
+        , user_email
+        , user_address_country
+    from mitxonline_users
+
+    union distinct
```
Comment (abeglova) [inline src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql:22]: This can just be union instead of union distinct

**Review by abeglova — APPROVED** [review]
LGTM other than the one small comment

File: src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql (hunk 1 above)
Comment (rachellougee) [reply src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql:22]: sqlfluff added it automatically when I just write union

**Review by abeglova — CHANGES_REQUESTED** [review]
Acutally, looks like you are not deduping between mitxonline users and edxorg users (and probably can't with the data we have). I don't really see the point of this pr then? Having them both in one table without deduping just makes joins harder

File: src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql (hunk 1 above)
Comment (abeglova) [reply src/ol_dbt/models/intermediate/mitx/int__mitx__users.sql:22]: Ok it's fine then

**Comment by rachellougee** [comment]
I guess the question is how do we dedup users between two platforms - MITx Online and edx.org? @abeglova @pdpinch  @Ferdi?

**Comment by abeglova** [comment]
For non-micromasters users we probably can't dedupe (other than by using email which is unreliable) but then why create this table?

**Review by abeglova — APPROVED** [review]
LGTM

**Comment by rachellougee** [comment]
Per discussion, we don't need to worry about deduplication right now as these are two different platforms. We will deal with duplicates later if any. We still want a model for MITx users, thus this PR is needed.

**Pull request merged**